wrapanapi/lenovo.py

```diff
@@ -357,5 +360,84 @@ def inventory(self, *requested_items, **kwargs):
         requested_items = requested_items or self._inventory_available
         return {item: self._inventory_available[item](self, requester) for item in requested_items}
 
+    def get_network_devices(self, server_name):
+        addin_cards = self.get_addin_cards(server_name) or []
+        pci_devices = self.get_pci_devices(server_name) or []
+        network_devices = []
+
+        for addin_card in addin_cards:
+            if (self.is_network_device(addin_card) and not
+                    self.is_device_in_list(addin_card, network_devices)):
+                network_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (self.is_network_device(pci_device) and not
+                    self.is_device_in_list(pci_device, network_devices)):
+                network_devices.append(pci_device)
+
+        return network_devices
+
+    def get_storage_devices(self, server_name):
+        addin_cards = self.get_addin_cards(server_name) or []
+        pci_devices = self.get_pci_devices(server_name) or []
+        storage_devices = []
+
+        for addin_card in addin_cards:
+            if (self.is_storage_device(addin_card) and not
+                    self.is_device_in_list(addin_card, storage_devices)):
+                storage_devices.append(addin_card)
+
+        for pci_device in pci_devices:
+            if (self.is_storage_device(pci_device) and not
+                    self.is_device_in_list(pci_device, storage_devices)):
+                storage_devices.append(pci_device)
+
+        return storage_devices
+
+    def is_device_in_list(self, device, device_list):
+        device_id = self.get_device_unique_id(device)
+
+        for d in device_list:
+            if device_id == self.get_device_unique_id(d):
+                return True
+
+        return False
+
+    def is_network_device(self, device):
+        device_name = device.get("productName") or device.get("name")
+        device_name = device_name.lower()
+
+        return (device.get("class") == "Network controller" or
```

Comment:
**Optional**

Consider making the comparison all in lowercase, to avoid changes to capitalization braking things. Below with `Mass storage controller` as well.